src/core/api/index.js

```diff
@@ -62,27 +63,35 @@ export function callApi({
 
   return fetch(apiURL, options)
     .then((response) => {
-      return response.clone().json()
-        .then(
-          (jsonResponse) => ({ response, jsonResponse }),
-          (error) => {
+      function makeTextResponseJson(textResponse) {
+        return { jsonResponse: { text: textResponse }, response };
+      }
+
+      if (response.headers.get('Content-Type') === 'application/json') {
+        return response.json()
+          .then((jsonResponse) => ({ response, jsonResponse }))
+          .catch((error) => {
             log.warn('Could not parse response as JSON:', error);
-            return response.clone().text().then((textResponse) =>
-              ({ response, jsonResponse: { text: textResponse } })
-            );
-          }
-        );
+
+            return response.text()
```

Comment:
Arg, fair point. Yeah the log will output that the JSON from the server was received but malformed, and then we can output "Unknown error, bad JSON" or whatnot 👍